theano/compile/function.py

```diff
@@ -185,6 +185,22 @@ def opt_log1p(node):
 
 
     """
+    if isinstance(outputs, dict):
```

Comment:
Update the docstring to tell that outputs can be a dictionary, and what the requirements (wrt sorting, etc.) are.